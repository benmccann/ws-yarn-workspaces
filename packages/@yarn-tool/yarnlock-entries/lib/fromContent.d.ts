/// <reference types="node" />
import { IYarnLockParsedV1, IYarnLockParsedV2 } from '@yarn-tool/yarnlock-parse/index';
export declare function fromContent<T extends IYarnLockParsedV1 | IYarnLockParsedV2>(yarnlock_old: Buffer | string): import("./YarnLockIterator").YarnLockIterator<T, import("@yarn-tool/yarnlock-parse/index").IUnpackYarnLockDataRow<T>>;
export declare function fromContentAsync<T extends IYarnLockParsedV1 | IYarnLockParsedV2>(yarnlock_old: Buffer | string): Promise<import("./YarnLockIterator").YarnLockIterator<T, import("@yarn-tool/yarnlock-parse/index").IUnpackYarnLockDataRow<T>>>;
export default fromContent;
