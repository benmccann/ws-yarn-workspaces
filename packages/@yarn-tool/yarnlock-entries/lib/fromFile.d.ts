import { PathLike } from "fs";
import { IYarnLockParsedV1, IYarnLockParsedV2 } from '@yarn-tool/yarnlock-types';
export declare function fromFile<T extends IYarnLockParsedV1 | IYarnLockParsedV2>(yarnlock_file: PathLike): import("./YarnLockIterator").YarnLockIterator<T, import("@yarn-tool/yarnlock-types").IUnpackYarnLockDataRow<T>>;
export declare function fromFileAsync<T extends IYarnLockParsedV1 | IYarnLockParsedV2>(yarnlock_file: PathLike): Promise<import("./YarnLockIterator").YarnLockIterator<T, import("@yarn-tool/yarnlock-types").IUnpackYarnLockDataRow<T>>>;
