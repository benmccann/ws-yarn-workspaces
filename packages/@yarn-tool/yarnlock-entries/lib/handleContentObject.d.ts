import { IYarnLockParsedV1, IYarnLockParsedV2 } from '@yarn-tool/yarnlock-parse/index';
import YarnLockIterator from './YarnLockIterator';
export declare function handleContentObject<T extends IYarnLockParsedV1 | IYarnLockParsedV2>(data: T): YarnLockIterator<T, import("@yarn-tool/yarnlock-parse").IUnpackYarnLockDataRow<T>>;
