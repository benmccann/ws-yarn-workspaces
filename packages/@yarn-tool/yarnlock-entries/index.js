"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.YarnLockIterator = exports.fromFile = exports.fromFileAsync = exports.fromContent = exports.fromContentAsync = void 0;
const YarnLockIterator_1 = require("./lib/YarnLockIterator");
Object.defineProperty(exports, "YarnLockIterator", { enumerable: true, get: function () { return YarnLockIterator_1.YarnLockIterator; } });
const fromContent_1 = require("./lib/fromContent");
Object.defineProperty(exports, "fromContentAsync", { enumerable: true, get: function () { return fromContent_1.fromContentAsync; } });
Object.defineProperty(exports, "fromContent", { enumerable: true, get: function () { return fromContent_1.fromContent; } });
const fromFile_1 = require("./lib/fromFile");
Object.defineProperty(exports, "fromFileAsync", { enumerable: true, get: function () { return fromFile_1.fromFileAsync; } });
Object.defineProperty(exports, "fromFile", { enumerable: true, get: function () { return fromFile_1.fromFile; } });
__exportStar(require("./lib/types"), exports);
exports.default = fromContent_1.fromContent;
//# sourceMappingURL=index.js.map