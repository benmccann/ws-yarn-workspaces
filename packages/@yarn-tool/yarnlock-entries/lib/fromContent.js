"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.fromContentAsync = exports.fromContent = void 0;
const index_1 = require("@yarn-tool/yarnlock-parse/index");
const handleContentObject_1 = require("./handleContentObject");
function fromContent(yarnlock_old) {
    return handleContentObject_1.handleContentObject(index_1.yarnLockParse(yarnlock_old));
}
exports.fromContent = fromContent;
async function fromContentAsync(yarnlock_old) {
    return fromContent(yarnlock_old);
}
exports.fromContentAsync = fromContentAsync;
exports.default = fromContent;
//# sourceMappingURL=fromContent.js.map