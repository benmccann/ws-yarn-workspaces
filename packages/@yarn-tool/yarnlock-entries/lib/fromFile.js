"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.fromFileAsync = exports.fromFile = void 0;
const tslib_1 = require("tslib");
const fs_1 = require("fs");
const fromContent_1 = (0, tslib_1.__importDefault)(require("./fromContent"));
function fromFile(yarnlock_file) {
    return (0, fromContent_1.default)((0, fs_1.readFileSync)(yarnlock_file));
}
exports.fromFile = fromFile;
async function fromFileAsync(yarnlock_file) {
    return fromFile(yarnlock_file);
}
exports.fromFileAsync = fromFileAsync;
//# sourceMappingURL=fromFile.js.map