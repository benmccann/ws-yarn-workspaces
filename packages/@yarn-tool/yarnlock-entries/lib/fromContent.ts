import { yarnLockParse } from '@yarn-tool/yarnlock-parse';
import { handleContentObject } from './handleContentObject';
import { IYarnLockParsedV1, IYarnLockParsedV2 } from '@yarn-tool/yarnlock-types';

export function fromContent<T extends IYarnLockParsedV1 | IYarnLockParsedV2>(yarnlock_old: Buffer | string)
{
	return handleContentObject<T>(yarnLockParse(yarnlock_old))
}

export async function fromContentAsync<T extends IYarnLockParsedV1 | IYarnLockParsedV2>(yarnlock_old: Buffer | string)
{
	return fromContent<T>(yarnlock_old)
}

export default fromContent;
