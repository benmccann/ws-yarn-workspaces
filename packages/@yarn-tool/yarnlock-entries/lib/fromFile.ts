import type { IYarnLockParsedV1, IYarnLockParsedV2 } from '@yarn-tool/yarnlock-parse/index';
import { PathLike, readFileSync } from "fs";
import fromContent from './fromContent';

export function fromFile<T extends IYarnLockParsedV1 | IYarnLockParsedV2>(yarnlock_file: PathLike)
{
	return fromContent<T>(readFileSync(yarnlock_file))
}

export async function fromFileAsync<T extends IYarnLockParsedV1 | IYarnLockParsedV2>(yarnlock_file: PathLike)
{
	return fromFile<T>(yarnlock_file)
}
