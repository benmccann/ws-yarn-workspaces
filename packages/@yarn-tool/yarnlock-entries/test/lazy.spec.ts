//@noUnusedParameters:false

import { basename, extname, join } from 'path';
import {
	detectYarnLockVersion,
	detectYarnLockVersionByFile,

} from '@yarn-tool/detect-yarnlock-version';
import { __TEST_YARNLOCK } from '../../../../__root_ws';
import { readFileSync } from 'fs';
import fromContent from '../lib/fromContent';
import { EnumDetectYarnLock } from '@yarn-tool/yarnlock-types';
import { _forEachVersionTags } from '../../../../test/lib/forEachVersionTags';
import { sync as FastGlob } from '@bluelovers/fast-glob';

beforeAll(async () =>
{

});

describe(basename(__filename, extname(__filename)), () =>
{

	_forEachVersionTags().forEach(ver =>
	{
		describe(ver, () =>
		{
			const dir = join(__TEST_YARNLOCK, ver);

			FastGlob([
				'**/*.lock',
			], {
				cwd: dir,
			}).forEach(name =>
			{
				const file = join(__TEST_YARNLOCK, ver, name);
				const buf = readFileSync(file);

				let actual = fromContent(buf);

				test(name, () =>
				{
					expect(actual.verType).toStrictEqual(EnumDetectYarnLock[ver]);

					if (ver === 'v1')
					{
						expect(actual.isV1()).toBeTruthy()
						expect(actual.isV2()).toBeFalsy()
					}
					else
					{
						expect(actual.isV2()).toBeTruthy()
						expect(actual.isV1()).toBeFalsy()
					}

					let output = actual.stringify();

					expect(detectYarnLockVersion(output)).toStrictEqual(EnumDetectYarnLock[ver]);

					expect(output.slice(0, 160)).toMatchSnapshot();
				});

				(ver !== 'v1') && test(`${name} check iterator`, () =>
				{
					for (const row of actual.iteratorRaw())
					{
						try
						{
							// @ts-ignore
							actual._normalize(row.raw, row.key);
						}
						catch (e)
						{
							console.dir(row);
							throw e
						}
					}
				})
			});
		});

	});

})
