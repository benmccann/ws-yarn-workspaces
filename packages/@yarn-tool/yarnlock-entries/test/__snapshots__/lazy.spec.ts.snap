// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lazy.spec v1 yarn.lock 1`] = `
"# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1


\\"@babel/code-frame@^7.0.0\\", \\"@babel/code-frame@^7.10.1\\":
  version \\"7.10.1"
`;

exports[`lazy.spec v2 meta-only.lock 1`] = `
"# This file is generated by running \\"yarn install\\" inside your project.
# Manual changes might be lost - proceed with caution!

__metadata:
  version: 4
"
`;

exports[`lazy.spec v2 yarn.lock 1`] = `
"# This file is generated by running \\"yarn install\\" inside your project.
# Manual changes might be lost - proceed with caution!

__metadata:
  version: 4

\\"libra"
`;

exports[`lazy.spec v3 meta-only.lock 1`] = `
"# This file is generated by running \\"yarn install\\" inside your project.
# Manual changes might be lost - proceed with caution!

__metadata:
  version: 6
  cache"
`;

exports[`lazy.spec v3 v4-tsdx.lock 1`] = `
"# This file is generated by running \\"yarn install\\" inside your project.
# Manual changes might be lost - proceed with caution!

__metadata:
  version: 7
  cache"
`;

exports[`lazy.spec v3 yarn.lock 1`] = `
"# This file is generated by running \\"yarn install\\" inside your project.
# Manual changes might be lost - proceed with caution!

__metadata:
  version: 6
  cache"
`;

exports[`lazy.spec v3 yarn-version-7.lock 1`] = `
"# This file is generated by running \\"yarn install\\" inside your project.
# Manual changes might be lost - proceed with caution!

__metadata:
  version: 7
  cache"
`;
